pkg/cfn/waiter/update_nodegroup.go

```diff
@@ -0,0 +1,69 @@
+package waiter
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	ekstypes "github.com/aws/aws-sdk-go-v2/service/eks/types"
+	"github.com/kris-nova/logger"
+
+	"github.com/weaveworks/eksctl/pkg/awsapi"
+)
+
+// WaitForNodegroupUpdate waits for an update to finish. Once it's done it returns the final status.
+// If the status was not Successful it will also return an error.
+func WaitForNodegroupUpdate(updateID string, api awsapi.EKS, timeout time.Duration, nextDelay NextDelay) (string, error) {
+	var lastStatus string
+	waiter := &Waiter{
+		NextDelay: nextDelay,
+		Operation: func() (bool, error) {
+			var (
+				err     error
+				success bool
+			)
+			lastStatus, success, err = describeUpdateStatus(context.Background(), updateID, api)
```

Comment:
Let's pass the `Context` as an argument to `WaitForNodegroupUpdate`. The call site already has a `Context` available.

Reply:
That thing needs a separate context, right? Like the other waiting thing implementation.

Reply:
Like the stack waiter:

```
// WaitForStack waits for the cluster stack to reach a success or failure state, and returns the stack.
func WaitForStack(ctx context.Context, cfnAPI awsapi.CloudFormation, stackID, stackName string, nextDelay NextDelay) (*types.Stack, error) {
	var lastStack *types.Stack
	waiter := &Waiter{
		NextDelay: nextDelay,
		Operation: func() (bool, error) {
			var (
				err     error
				success bool
			)
			lastStack, success, err = describeStackStatus(context.Background(), cfnAPI, stackID, stackName)
			return success, err
		},
	}

```

I assumed it's for a reason. Maybe because it doesn't want the context to be cancelled. Also, I'm calling WaitWithTimeout which doesn't take a context at all anyways. :)

Reply:
That said, for the stack it might be an oversight. So I guess I can pass it along for the DescribeUpdate call ¯\\\_(ツ)\_/¯